tiledb/sm/query/writer.cc

```diff
@@ -1304,9 +1309,14 @@ Status Writer::check_subarray() const {
     if (layout_ == Layout::UNORDERED && subarray_.is_set())
       return LOG_STATUS(Status::WriterError(
           "Cannot initialize query; Setting a subarray in unordered writes for "
-          "dense arrays in inapplicable"));
+          "dense arrays is inapplicable"));
+  } else {  // state is !array_schema_->dense()
```

Comment:
remove the comment and put this in an `assert()`

Reply:
done